src/main/scala/sbtlicensereport/SbtLicenseReport.scala

```diff
@@ -68,6 +83,40 @@ object SbtLicenseReport extends AutoPlugin {
           streams.value.log
         )
       },
+      updateLicensesAggregate := {
```

Comment:
Is this needed? I am wondering if dumpLisenseReportAggregate can be a dynamic task that calls updateLisenses using the filter defined within the dynamic task?

Reply:
So the reason I made this task is because `updateLicenses` exists and it appears that there is an expectation that users publicly call this setting within sbt outside of the scope of `dumpLisenceReport` (otherwise the `updateLicenses` wouldn't exist as a setting within the first place)?

Reply:
@eed3si9n I think I understand what you are trying to say now. You want `updateLicenses` to a general purpose setting that retrieves **all** of the dependencies of the project and then the filtering is done in the various `dump*` tasks?

If so then at least to me that makes more sense, my only counter argument here is that as a user you lose visibility about where the licenses come from. The various `dump*` tasks merge and collate everything which means it can be a bit hard to figure out which licenses come from which sbt projects where as the having a specific setting for each filter makes it clear where its coming from.

However if you feel strongly on this (and I am correct in your intention) let me know and I will change it.

Reply:
Honestly I don't know what is expected of this plugin, and to certain extent we can probably make things up to fit the needs of current users/contributors. A situation where having programmatic access to license report type thing would be useful is when you want to upload it to some other compliance software, like Mend.io.

I was thinking more like, per-subproject you run `updateLicenses` (analogous to `update`?), and have `dumpLicenseReportAggregate` that does the aggregation using dynamic task without `updateLicensesAggregate` intermediaries.

Reply:
So the core design behind my implementation I took from https://github.com/albuch/sbt-dependency-check/ which as you can tell is one of the more popular sbt plugins and fills the exact same role as this plugin, just that instead of making LICENSE reports it makes security reports instead and in that specific plugin it uses filters (i.e. `.all(ScopeFilter(inAggregates(thisProjectRef.value)))` rather than having a task per project to (i.e. your suggestion for `updateLicenses`).

From what I can tell the reason behind this design is usage, i.e. in almost every case these projects are designed to make a report of the whole sbt build for legal/compliance/process reasons and hence which sub projects happen to have which licenses is not really important (unlike `update` where its **very** important for obvious reasons) aside for possible debugging reasons.

That being said, the existence of `updateLicenses` is something that sbt-dependency-check [doesn't have](https://github.com/albuch/sbt-dependency-check/blob/a121bfb29a6412bd6cfe0cf15a299a084baa77fb/src/main/scala/net/vonbuchholtz/sbt/dependencycheck/DependencyCheckKeys.scala#L128-L133), it just uses the filters directly and even the naming of `updateLicenses` is somewhat confusing (I am not surprised that judging from the name alone, you expect it to work like normal `update` does).

So I guess one strategy (which is mine) is to make this plugin's design mirror sbt-dependency-check for the reasons provided earlier in which case there is a strong argument to just remove `updateLicenses` to prevent confusion. Another strategy would be to do as what you say, which is make `updateLicenses` behave properly on a per project basis and then have `dumpLicenseReportAggregate` a dyanmic task that collects all of the project licenses. In doing so there may be merit in upstreaming a similar change to sbt-dependency-check.

wdyt?

Reply:
Whether the subproject-level granularity matters or not I think is case by case. Like I said having `updateLicenses` allows programatic access to the information, so I'd be against the removal of that. Whether we want to have lots of `updateLicense*` or `dumpLicenseReport*` is up to you. My suggestion on using the dynamic task really is just a suggestion, and it's non-blocking.

Reply:
Im convinced, I will remove the other `updateLicenses*` and figure out how to simulate the same behaviour in a dynamic task. Will also consider doing a similar change to sbt-dependency-check if they are open to it.

Reply:
@eed3si9n  Okay so I just pushed the changes, turned out it was a lot more trivial than expected and I was overthinking it.